### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by Jfivvf**

### CODE REVIEW

File: homework_(32) (1).ipynb [hunk 1]
```diff
@@ -0,0 +1,1941 @@
+{
```
Comment (Pretzandalf) [inline homework_(32) (1).ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #52.            attn_probs = F.softmax(attn_scores, dim=-1)
</pre><p>От себя на будущие добавлю, старайся использовать nn вместо F. Функции в nn содержат под капотом те же F функции, но они сделаны так чтобы удобнее работать с обучаемыми параметрами и наример с батчами. Скажем, так что nn более высокоуровневый. (здесь вообще не важно nn или F, просто как фанфакт))</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Jfivvf/summer_ML/pull/1/discussion/#comment-2192945025'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (Pretzandalf) [inline homework_(32) (1).ipynb:1]: <p>отлично сделанная последовательность дропаутов:</p>

<pre class="ql-syntax" spellcheck="false">norm(x + dropout(sublayer(x))
</pre><p>в трансформерах используется именно такая последовательность</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Jfivvf/summer_ML/pull/1/discussion/#comment-2192945030'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (Pretzandalf) [inline homework_(32) (1).ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #53.            K_proj = torch.einsum('hkl,bhld-&gt;bhkd', e_proj, k)
</pre><p>это можно было сделать иначе, загляни в эталонный код от Дани.</p><p>Ты создаешь Е и F как линейные слои (что и есть матрицы), и потом просто перемножаешь нужные матрицы через оператор @ (матричное умножение) , получая ЕК и FV соответственно</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Jfivvf/summer_ML/pull/1/discussion/#comment-2192945032'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='79'/>

Comment (Pretzandalf) [inline homework_(32) (1).ipynb:1]: <p>Вся домашка сделано очень даже хорошо.</p><p>Молодец!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Jfivvf/summer_ML/pull/1/discussion/#comment-2192945037'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='93'/>
